packages/discovery-react-components/src/components/DocumentPreview/components/HtmlView/HtmlView.tsx

```diff
@@ -53,13 +55,46 @@ export const HtmlView: FC<Props> = ({
   }
 
   const [html, setHtml] = useState<string | null>(null);
+  const [processedDoc, setProcessedDoc] = useState<ProcessedDoc | null>(null);
+  const [locationArray, setLocationArray] = useState<Array<Location> | null>(null);
+
+  const textMappingBbox = usePassage(document, highlight as QueryResultPassage);
+  useEffect(() => {
+    if (highlight) {
+      if (isPassage(highlight)) {
+        if (processedDoc && textMappingBbox) {
+          let processedDocBbox: ProcessedBbox[] = [];
+          textMappingBbox &&
+            processedDoc.bboxes &&
+            textMappingBbox.forEach(Bbox => {
+              const tempBboxs = findMatchingBbox(Bbox, processedDoc.bboxes as ProcessedBbox[]);
+              processedDocBbox = processedDocBbox.concat(tempBboxs);
+            });
+          let passageLocs: Array<Location> = [];
+          processedDocBbox &&
+            processedDocBbox.forEach(Bbox => {
+              passageLocs.push(Bbox.location);
+            });
+          setLocationArray(passageLocs);
+        }
+      } else {
+        const tableLoc = get(highlight, 'table.location', {});
+        setLocationArray([tableLoc]);
+      }
+    }
+  }, [document, highlight, processedDoc, textMappingBbox]);
```

Comment:
Since this `useEffect` depends on `processedDoc` which is set by the following `useEffect`, swap their locations. This will make it read better top-to-bottom.